src/textual/widgets/_directory_tree.py

```diff
@@ -91,18 +94,58 @@ def __init__(
         classes: str | None = None,
         disabled: bool = False,
     ) -> None:
-        str_path = os.fspath(path)
-        self.path = str_path
+        """Initialise the directory tree.
+
+        Args:
+            path: Path to directory.
+            name: The name of the widget, or None for no name.
+            id: The ID of the widget in the DOM, or None for no ID.
+            classes: A space-separated list of classes, or None for no classes.
+            disabled: Whether the directory tree is disabled or not.
+        """
+        self.path = path
         super().__init__(
-            str_path,
-            data=DirEntry(str_path, True),
+            str(path),
+            data=DirEntry(Path(path)),
             name=name,
             id=id,
             classes=classes,
             disabled=disabled,
         )
 
-    def process_label(self, label: TextType):
+    def reload(self) -> None:
+        """Reload the `DirectoryTree` contents."""
+        self.reset(str(self.path), DirEntry(Path(self.path)))
+        self._load_directory(self.root)
+
+    def validate_path(self, path: str | Path) -> Path:
+        """Ensure that the path is of the `Path` type.
+
+        Args:
+            path: The path to validate.
+
+        Returns:
+            The validated Path value.
+
+        Note:
+            The result will always be a Python `Path` object, regardless of
+            the value given.
+        """
+        return Path(path)
```

Comment:
Almost. You do get a fresh object.  I suppose this could be modified to be:

```python
return path if isinstance(path, Path) else Path(path)
```

but that didn't seem to save much of anything.

Reply:
I think `return Path(path)` is a good idea to make sure that we don't get a path that the user mutates later.

Reply:
Fairly sure `Path` is immutable that that shouldn't be an issue.

Reply:
Yeah, I checked. Paths are indeed immutable.